lib/views/main.dart

```diff
@@ -83,8 +84,9 @@ class _NestedRouteObserver extends RouteObserver<PageRoute> {
   void didPush(Route<dynamic> route, Route<dynamic>? previousRoute) {
     super.didPush(route, previousRoute);
 
-    if (route.settings.arguments is Function) {
-      MainViewModel.appBar = (route.settings.arguments as Function)();
+    if (route.settings.arguments is NavigationArguments) {
+      MainViewModel.appBar =
```

Comment:
The old functionality was necessary to create a new AppBar since otherwise the listener would be registered multiple times on the same object, so that a dialog would be opened multiple times on adjusting the filter after navigating in the app. Will this work with the new implementation?

Reply:
It seems to work.